From 7be019bd20143d1e8ed69af5d895c0e3842c107f Mon Sep 17 00:00:00 2001
From: Alyssa Ross <hi@alyssa.is>
Date: Sat, 29 Jul 2023 12:03:18 +0000
Subject: [PATCH 4/5] vhost_user: renumber SHARED_MEMORY_REGIONS
SPDX-FileCopyrightText: 2019-2021 Alibaba Cloud. All rights reserved.
SPDX-FileCopyrightText: The Chromium OS Authors
SPDX-FileCopyrightText: 2023 Alyssa Ross <hi@alyssa.is>
SPDX-License-Identifier: Apache-2.0

0x0002_0000 is now standardized as VHOST_USER_PROTOCOL_F_XEN_MMAP[1].
Renumber SHARED_MEMORY_REGIONS to an available value to avoid
interoperability problems with other vhost-user implementations.

(It would be nice if VHOST_USER_PROTOCOL_F_SHARED_MEMORY was
standardised, to avoid this problem in future!)

[1]: https://gitlab.com/qemu-project/qemu/-/commit/f21e95ee97d5adb4a274b48c6c8f70a221c1f513

TEST=run crosvm vhost-user-gpu device with cross domain

Change-Id: I56c36e5ef75b68fd4244dee1206a2cd1bee7c452
Reviewed-on: https://chromium-review.googlesource.com/c/crosvm/crosvm/+/4727935
Reviewed-by: Daniel Verkamp <dverkamp@chromium.org>
Commit-Queue: Daniel Verkamp <dverkamp@chromium.org>
(cherry-picked from commit 6795b3edf3ddc0b3a2e2b928033f21dabcf8be78)
Signed-off-by: Alyssa Ross <hi@alyssa.is>
---
 vhost/src/vhost_user/message.rs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vhost/src/vhost_user/message.rs b/vhost/src/vhost_user/message.rs
index b02b925..9950c2f 100644
--- a/vhost/src/vhost_user/message.rs
+++ b/vhost/src/vhost_user/message.rs
@@ -448,10 +448,10 @@ bitflags! {
         const STATUS = 0x0001_0000;
         /// Support Xen mmap.
         const XEN_MMAP = 0x0002_0000;
-        /// Support shared memory regions.
-        const SHARED_MEMORY_REGIONS = 0x0002_0000;
         /// Support transferring internal device state.
         const DEVICE_STATE = 0x0008_0000;
+        /// Support shared memory regions. (Non-standard.)
+        const SHARED_MEMORY_REGIONS = 0x8000_0000;
     }
 }
 
-- 
2.44.0

